src/scalaDebugger.ts

```diff
@@ -46,15 +50,169 @@ export async function start(
 
 class ScalaConfigProvider implements vscode.DebugConfigurationProvider {
   provideDebugConfigurations(): ProviderResult<DebugConfiguration[]> {
-    return [];
+    const mainClassPick = "Main Class";
+    const testClassPick = "Test Suite";
+
+    return vscode.window
+      .showQuickPick([mainClassPick, testClassPick], {
+        placeHolder: "Pick kind of class to debug",
+      })
+      .then((result) => {
+        if (result === mainClassPick) {
+          return this.provideDebugMainClassConfiguration().then((config) => {
+            if (config === undefined) return [];
+            return [config];
+          });
+        } else if (result === testClassPick) {
+          return this.provideDebugTestClassConfiguration().then((config) => {
+            if (config === undefined) return [];
+            return [config];
+          });
```

Comment:
if we start using `Promise.reject` (see my comment below) this can be:

```suggestion
          return this.provideDebugTestClassConfiguration().then((config) => [config])
```

and then after the most outer `then` we can add

```ts
.catch(() => [])
```

Reply:
Thanks for the `Promise.reject` advice! I miss experience in JS :)
Since it's `Thenable`, it misses `catch` method, but have second argument callback `onrejected`.